syn/python/get_kge.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Parse a yosys area report and give a kGE equivalient
+
+import argparse
+
+
+def read_lib(lib_file_path):
+    with open(lib_file_path, 'r') as f:
+        lib_file = f.readlines()
+
+    cell_dict = {}
+    weighted_dict = {}
+    cell_name = ''
+    for line_idx, line in enumerate(lib_file):
+        if line.startswith('  cell ('):
+            if cell_name != '':
+                raise RuntimeError('Found cell while searching for area at \
+                    line ' + str(line_idx + 1))
```

Comment:
This is fine, but it's probably easier to write with the format function:
```python3
raise RuntimeError('Found cell while searching for area at line {}.'.format(1 + line_idx))
```

Conveniently, Python works like C and pastes adjacent string literals, so you can do things like this to break lines:
```python3
raise RuntimeError('This is my rather {} string, which '
                   'I am going to break to avoid long '
                   '{}.'
                   .format('long', 'lines'))
```

Also, I'd consider generating errors like a compiler "foo.txt:23: Oh no, this looks malformed" (you have the path, so it's not too painful to construct the message).

Reply:
Thanks - was wondering about line breaks. Will add the compiler-style messages too.